Error in Google Play services

```
type: crash
osVersion: google/raven/raven:13/TP1A.221105.002/2022111800:user/release-keys
package: com.google.android.gms:224459044
process: com.google.android.gms.persistent
processUptime: 1952045 + 309 ms
GmsCompatConfig version: 16

java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.wifi.CONFIGURED_NETWORKS_CHANGE flg=0x4000010 (has extras) } in com.google.android.gms.wearable.node.WifiService$SavedNetworkBroadcastReceiver@440d5a5
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$android-app-LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1800)
	at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(Unknown Source:2)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7904)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
Caused by: java.lang.SecurityException: WifiService: Neither user 10127 nor current process has android.permission.READ_WIFI_CREDENTIAL.
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3021)
	at android.os.Parcel.createException(Parcel.java:3005)
	at android.os.Parcel.readException(Parcel.java:2981)
	at android.os.Parcel.readException(Parcel.java:2923)
	at android.net.wifi.IWifiManager$Stub$Proxy.getPrivilegedConfiguredNetworks(IWifiManager.java:2210)
	at android.net.wifi.WifiManager.getPrivilegedConfiguredNetworks(WifiManager.java:1697)
	at ccsl.e(:com.google.android.gms@224459044@22.44.59 (190400-488547955):4)
	at ccsl.i(:com.google.android.gms@224459044@22.44.59 (190400-488547955):3)
	at ccsl.g(:com.google.android.gms@224459044@22.44.59 (190400-488547955):1)
	at com.google.android.gms.wearable.node.WifiService$SavedNetworkBroadcastReceiver.a(:com.google.android.gms@224459044@22.44.59 (190400-488547955):6)
	at com.google.android.gms.libs.punchclock.tracing.TracingBroadcastReceiver.onReceive(:com.google.android.gms@224459044@22.44.59 (190400-488547955):2)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$android-app-LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1790)
	... 11 more
Caused by: android.os.RemoteException: Remote stack trace:
	at android.app.ContextImpl.enforce(ContextImpl.java:2285)
	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:2313)
	at android.content.ContextWrapper.enforceCallingOrSelfPermission(ContextWrapper.java:948)
	at com.android.server.wifi.WifiServiceImpl.enforceReadCredentialPermission(WifiServiceImpl.java:1070)
	at com.android.server.wifi.WifiServiceImpl.getPrivilegedConfiguredNetworks(WifiServiceImpl.java:3029)


```

--- a/core/java/com/android/internal/gmscompat/PlayStoreHooks.java
+++ b/core/java/com/android/internal/gmscompat/PlayStoreHooks.java
@@ -103,7 +103,7 @@ public final class PlayStoreHooks {
                     PendingIntent.FLAG_CANCEL_CURRENT |
                         PendingIntent.FLAG_MUTABLE);
 
-            context.registerReceiver(sf, new IntentFilter(intentAction));
+            context.registerReceiver(sf, new IntentFilter(intentAction), Context.RECEIVER_NOT_EXPORTED);
             return sf.pendingIntent;
         }
 

--- a/core/java/com/android/internal/gmscompat/StubDef.java
+++ b/core/java/com/android/internal/gmscompat/StubDef.java
@@ -194,13 +194,23 @@ public class StubDef implements Parcelable {
                 Class t = parcelListType;
                 if (t == List.class) {
                     p.writeList(Collections.emptyList());
-                } else if (t == ParceledListSlice.class) {
-                    p.writeTypedObject(ParceledListSlice.emptyList(), 0);
-                } else if (t == StringParceledListSlice.class) {
-                    p.writeTypedObject(StringParceledListSlice.emptyList(), 0);
                 } else {
-                    Log.d(TAG, "unknown parcel list type " + type);
-                    return false;
+                    String listTypeName = t.getName();
+                    // There is android.content.pm.ParceledListSlice and
+                    // com.android.modules.utils.ParceledListSlice.
+                    // Moreover, when the latter is used in an APEX, it's prefixed like this:
+                    // com.android.wifi.x.com.android.modules.utils.ParceledListSlice
+
+                    // Same applies to StringParceledListSlice.
+
+                    if (listTypeName.endsWith(".ParceledListSlice")) {
+                        p.writeTypedObject(ParceledListSlice.emptyList(), 0);
+                    } else if (listTypeName.endsWith(".StringParceledListSlice")) {
+                        p.writeTypedObject(StringParceledListSlice.emptyList(), 0);
+                    } else {
+                        Log.d(TAG, "unknown parcel list type " + listTypeName);
+                        return false;
+                    }
                 }
                 break;
             }

